packages/core/src/credential/Credential.ts

```diff
@@ -292,118 +277,114 @@ export function fromClaim(
-  const ctypeMismatch = credential.claim.cTypeHash !== cTypeHash
-  const delegationMismatch = credential.delegationId !== delegationId
-  if (ctypeMismatch || delegationMismatch) {
-    throw new SDKErrors.CredentialUnverifiableError(
-      `Some attributes of the on-chain attestation diverge from the credential: ${[
-        'cTypeHash',
-        'delegationId',
-      ]
-        .filter((_, i) => [ctypeMismatch, delegationMismatch][i])
-        .join(', ')})}`
-    )
-  }
-  if (revoked && allowRevoked !== true) {
+  const attestation = Attestation.fromChain(
+    maybeAttestation,
+    credential.rootHash
+  )
+  Attestation.verifyAgainstCredential(attestation, credential)
+  const { owner: attester, revoked } = attestation
+  if (revoked && throwOnRevoked !== false) {
     throw new SDKErrors.RevokedTypeError('Attestation revoked')
-  } else if (typeof allowedAuthorities === 'undefined') {
-    return { attester, revoked, matchedTrustPolicy: {} }
-  } else if (allowedAuthorities[attester]?.isAttester === true) {
-    return { attester, revoked, matchedTrustPolicy: { isAttester: attester } }
-  } else if (credential.delegationId) {
-    const trustedDelegators: DidUri[] = []
-    Object.entries(allowedAuthorities).forEach(([did, trust]) => {
-      if (trust.isDelegator === true) {
-        trustedDelegators.push(did as DidUri)
-      }
-    })
-    if (trustedDelegators.length > 0) {
-      const delegation = await DelegationNode.fetch(credential.delegationId)
-      const { node } = await delegation.findAncestorOwnedBy(trustedDelegators)
-      if (node) {
-        return {
-          attester,
-          revoked,
-          matchedTrustPolicy: { isDelegator: node.account },
-        }
-      }
-    }
   }
-  throw new SDKErrors.CredentialUnverifiableError(
-    'This attestation does not match any given trust policy and thus is not trusted'
-  )
+  return { attester, revoked }
 }
 
-type VerifyOptions = {
-  ctype?: ICType
-  challenge?: string
-  didResolveKey?: DidResolveKey
-  allowedAuthorities?: TrustPolicies
-  allowRevoked?: boolean
+export interface VerifiedCredential extends ICredential {
+  revoked: boolean
+  attester: DidUri
+}
+
+/**
+ * Updates the revocation status of a previously verified credential to allow checking if it is still valid.
+ *
+ * @param verifiedCredential The output of [[verifyCredential]] or [[verifyPresentation]], which adds a `revoked` and `attester` property.
+ * @param throwOnRevoked If not set to false, a revoked attestation will result in a rejected Promise.
+ * @returns A promise of resolving to the same object but with the `revoked` property updated. The promise rejects if the attestation has been deleted or its data changed since verification.
+ */
+export async function recheckRevocationStatus(
+  verifiedCredential: VerifiedCredential,
+  throwOnRevoked = true
+): Promise<VerifiedCredential> {
+  const { revoked, attester } = await verifyAttested(
+    verifiedCredential,
+    throwOnRevoked
```

Comment:
Do we not want this always to be the case? We decided that if there was an error, we'd always throw it.

Is there a reason why we might not?